spec/controllers/orders_controller_spec.rb

```diff
@@ -11,16 +11,31 @@
 
   describe "GET #index" do
     it "gets index" do
-      get :index
       sign_in user
+      get :index
+
       response.should be_success
     end
   end
 
-  describe "GET show" do
+  describe "GET #index as a visitor not logged in " do
+    it "get a 401 redirection if visitor is not logged in" do
+      get :index
+      expect(response).to have_http_status(:found)
+    end
+
+    it "redirect user to devise sign in" do
+      get :index
+      expect(response).to redirect_to("http://test.host/users/sign_in")
```

Comment:
tu peux et devrais changer par un relative path :

... to redirect_to new_user_session_path

Reply:
Ca te permet de jamais avoir d'erreur si tu changes des hosts ou autres.

Reply:
Oui j'ai essayais mais ça ne passait pas. 
Je recommencerais ok.